src/stavisky_lab_to_nwb/stavisky/staviskydecodinginterface.py

```diff
@@ -0,0 +1,166 @@
+
+    def __init__(
+        self,
+        port: int,
+        host: str,
+    ):
+        super().__init__(port=port, host=host)
+
+    def add_to_nwbfile(
+        self,
+        nwbfile: NWBFile,
+        metadata: dict,
+        smooth_timestamps: bool = False,  # TODO: removal/modification pending timestamp meeting
+    ):
+        # initialize redis client and check connection
+        r = redis.Redis(
+            port=self.source_data["port"],
+            host=self.source_data["host"],
+        )
+        r.ping()
+
+        # get processing module
+        module_name = "decoding"
+        module_description = "Contains decoder outputs for real-time speech decoding"
+        processing_module = get_module(nwbfile=nwbfile, name=module_name, description=module_description)
+
+        # get reference time for timestamps
+        # session_start_time = metadata["NWBFile"]["session_start_time"].timestamp()
+        session_start_time = np.frombuffer(r.xrange("metadata")[0][1][b"startTime"], dtype=np.float64).item()
+
+        # loop through data and make arrays
+        decoder_output = r.xrange("binned:decoderOutput:stream")  # TODO: read in chunks?
+        logits = []
+        timestamps = []
+        for entry in decoder_output:
+            # append data until end of each trial
+            if b"start" in entry[1].keys():  # start of trial decoding period
+                trial_logits = []
+            elif b"end" in entry[1].keys():  # end of trial decoding period
+                if smooth_timestamps:  # interpolate timestamps to be regular, if desired
+                    start_time = int(trial_logits[0][0].split(b"-")[0]) / 1000.0 - session_start_time
+                    stop_time = int(trial_logits[-1][0].split(b"-")[0]) / 1000.0 - session_start_time
+                    trial_timestamps = np.linspace(start_time, stop_time, len(trial_logits - 2))
+                else:
+                    trial_timestamps = np.array(
+                        [(int(tl[0].split(b"-")[0]) / 1000.0 - session_start_time) for tl in trial_logits]
+                    )
+                timestamps.append(trial_timestamps)
+
+                trial_logits = np.stack(
+                    [np.frombuffer(tl[1][b"logits"], dtype=np.float32) for tl in trial_logits], axis=0
+                )
+                logits.append(trial_logits)
+                trial_logits = []  # reset data stack
+            elif b"logits" in entry[1].keys():
+                trial_logits.append(entry)
+        logits = np.concatenate(logits, axis=0)
+        timestamps = np.concatenate(timestamps)
+
+        # create timeseries obj
+        logits_timeseries = TimeSeries(
+            name="phoneme_logits",
+            data=logits,
+            unit="n.a.",
+            timestamps=timestamps,
+            description="Log-probabilities of the 39 phonemes plus silence and space between words, as "
+            + "predicted by an RNN decoder",
```

Comment:
Usual way of breaking this up stylistically with `black` is as follows

```suggestion
            description=(
                "Log-probabilities of the 39 phonemes plus silence and space between words, as "
                "predicted by an RNN decoder"
            )
```

Reply:
triple-quotes are also accepted (stylistically and syntactically) but can look a little more awkward because the block wraps to the left edge after the first line